adapters/madsense/madsense.go

```diff
@@ -0,0 +1,126 @@
+package madsense
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the MadSense adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	reqs := make([]*adapters.RequestData, 0, len(request.Imp))
+	var errs []error
+
+	appendReq := func(imps []openrtb2.Imp) {
+		req, err := a.makeRequest(request, imps)
+		if err != nil {
+			errs = append(errs, err)
+			return
+		}
+		if req != nil {
+			reqs = append(reqs, req)
+		}
+	}
+
+	var videoImps, bannerImps []openrtb2.Imp
+	for _, imp := range request.Imp {
```

Comment:
Here as well, you appear to make a copy of the slice, perhaps you can use a reference to avoid the copy?
```suggestion
	for i := range request.Imp {
		imp := &request.Imp[i]
```

Reply:
done